backend/models/customer.js

```diff
@@ -3,15 +3,42 @@ const ClientBase = require("./client");
 
 const Schema = mongoose.Schema;
 
+var address = {
+  addressName: { type: String },
+  name: { type: String },
+  addressLine: { type: String },
+  city: { type: String },
+  state: { type: String },
+  zipCode: { type: String },
+  phone: { type: String },
+};
+
+var shoppingList = {
+  title: { type: String },
+  wishedProducts: [],
+};
+
+var shoppingCart = {
+  productsIn: { type: String },
+  productAmount: { type: Number },
+};
+
+var creditCard = {
+  creditCardNumber: { type: String },
+  creditCardCvc: { type: String },
+  creditCardData: { type: String },
+  creditCardName: { type: String },
+};
+
 var customerSchema = new Schema(
   {
-    shoppingLists: { type: String },
-    orders: { type: String },
-    cart: { type: String },
-    addresses: { type: String },
+    shoppingLists: [shoppingList],
+    orderIds: [],
```

Comment:
Presence of an order model makes this `orderIds` field redundant.

Reply:
Removed.